coriolis_openstack_utils/actions/flavor_actions.py

```diff
@@ -0,0 +1,169 @@
+        " '%s'.")
+
+    @property
+    def flavor_name_format(self):
+        return CONF.destination.new_flavor_name_format
+
+    def check_same_flavor(self, src_flavor, dest_flavor):
+        relevant_keys = ['ram', 'disk', 'vcpus', 'rxtx_flavor']
+        dest_flavor_info = {k: v for k, v in dest_flavor.items()
+                            if k in relevant_keys}
+        src_flavor_info = {k: v for k, v in src_flavor.items()
+                           if k in relevant_keys}
+
+        if (src_flavor_info == dest_flavor_info and
+                dest_flavor['name'] == self.get_new_flavor_name()):
+            return True
+
+        return False
+
+    def check_already_done(self):
+        dest_flavor_name = self.get_new_flavor_name()
+
+        src_flavor_id = self.payload['src_flavor_id']
+        src_flavor = self._source_openstack_client.nova.flavors.get(
+            src_flavor_id)
+
+        dest_flavor_name = self.get_new_flavor_name()
+        dest_flavor = None
+        try:
+            dest_flavor = self._destination_openstack_client.nova.flavors.find(
+                name=dest_flavor_name, is_public=None)
+            if self.check_same_flavor(src_flavor.to_dict(),
+                                      dest_flavor.to_dict()):
+                return {"done": True, "result": dest_flavor.to_dict()}
+            else:
+                raise Exception("Found destination flavor named \"%s\", but "
+                                "with different properties than source flavor "
+                                "\"%s\" !" %
+                                (dest_flavor_name, src_flavor_id))
+
+        except novaclient.exceptions.NotFound:
+            return {"done": False, "result": None}
+
+    def equivalent_to(self, other_action):
+        if other_action.action_type == self.action_type:
+            if self.payload['src_flavor_id'] == (
+                    other_action.payload.get('src_flavor_id')):
+                return True
+        return False
+
+    def print_operations(self):
+        super(FlavorCreationAction, self).print_operations()
+        dest_flavor_name = self.get_new_flavor_name()
+        LOG.info(
+            "Create new destination flavor named \"%s\" with sources' "
+            "properties." % (dest_flavor_name))
+
+    def get_new_flavor_name(self):
+        return self.flavor_name_format % {
+            "original": self._source_openstack_client.nova.flavors.get(
+                self.payload['src_flavor_id']).name}
+
+    def create_flavor_body(self):
+        src_flavor = self._source_openstack_client.nova.flavors.get(
+            self.payload['src_flavor_id'])
+        relevant_keys = ['ram', 'disk', 'vcpus', 'swap', 'rxtx_factor']
+        src_flavor_info = src_flavor.to_dict()
+        body = {k: v for k, v in src_flavor_info.items()
+                if k in relevant_keys}
+        nova_client = self._destination_openstack_client.nova
+        if nova_client.api_version.get_string() >= '2.55':
```

Comment:
a) double-check that the Nova client automatically uses the latest microversion (and doesn't default to '2.0' or whatever
b) either cast both to floats or use `oslo_version` (not `oslo_versionedobject`) and then do the comparison (do not compare version strings unless they're timestamps or something with a fixed structure!)